code/types.ts

```diff
@@ -0,0 +1,26 @@
+import { APIGatewayProxyEvent, APIGatewayProxyResult, Context } from "aws-lambda";
+
+export type ApiGatewayProxyHandler = (event: APIGatewayProxyEvent, context: Context) => Promise<APIGatewayProxyResult>
+
+export enum HttpStatus {
+    OK = 200,
+    NoContent = 204,
+    BadRequest = 400,
+    NotFound = 404,
+    ServerError = 500,
+}
+
+export interface MethodHandlerMap {
+    [method: string]: ApiGatewayProxyHandler
+}
+
+export interface ResourceHandlerMap {
```

Comment:
When defining this kind of objects you can use the Record type, for example:
export type ResourceHandlerMap = Record<string, MethodHandlerMap>